File: packages/javascript/lib/word/randPlant.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Plants: trees, flowers, leaves and what grows on them.
 *
 * `randWord({ theme: 'plant' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randPlant({ language: 'ko', count: 3 }); // ['나무', '민들레', '꽃']
 * randPlant({ language: 'en', count: 3 }); // ['Treetop', 'Blossom', 'Fern']
 */
export function randPlant(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Plant words with the language and theme each one came from.
 *
 * @example
 * randPlant({ language: 'ko', output: 'detail' });
 * // [{ word: '나무', language: 'ko', theme: 'plant' }]
 */
export function randPlant(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randPlant(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('plant', options);
}

File: packages/dart/lib/src/word/rand_plant.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Plants: trees, flowers, leaves and what grows on them.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.plant] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randPlant(language: WordLanguage.ko, count: 3); // [나무, 민들레, 꽃]
/// randPlant(language: WordLanguage.en, count: 3); // [Treetop, Blossom, Fern]
/// ```
List<String> randPlant({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.plant,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
